int32[2] size
int32 num
int32[] index
float64[] val